codec/proto.go

```diff
@@ -93,8 +94,8 @@ func (c Codec) Encode(w io.Writer, i interface{}) error {
 	return nil
 }
 
-// Decode method.
-func (c Codec) Decode(r io.Reader) (interface{}, error) {
+// Decode will decode the binary data of the given reader into the original entity.
```

Comment:
Encode mentions the error cases possibly created by the user and not the network. None of these should ever happen on decode, as it would already have failed on encode. That's why there is a difference.